receiver/sapnetweaverreceiver/scraper.go

```diff
@@ -223,3 +228,75 @@ func (s *sapNetweaverScraper) collectGetAlertTree(_ context.Context, now pcommon
 	s.recordSapnetweaverSessionsBrowserCountDataPoint(now, alertTreeResponse, errs)
 	s.recordSapnetweaverSessionsEjbCountDataPoint(now, alertTreeResponse, errs)
 }
+
+// collectCertificateValidity collects certificate final validity date.
+func (s *sapNetweaverScraper) collectCertificateValidity(_ context.Context, now pcommon.Timestamp, errs *scrapererror.ScrapeErrors) {
+	args := []string{"-L", "/usr/sap", "-name", "*.pse"}
+	certFilesPaths, err := s.service.FindFile(args...)
+	if err != nil {
+		errs.AddPartial(1, fmt.Errorf("failed to find certificate files: %w", err))
+		return
+	}
+
+	for _, certFilePath := range certFilesPaths {
+		if certFilePath == "" {
+			continue
+		}
+
+		command := fmt.Sprintf("/usr/sap/hostctrl/exe/sapgenpse get_my_name -p %s -n validity", certFilePath)
+		resp, err := s.service.OSExecute(command)
+		if err != nil {
+			errs.AddPartial(1, fmt.Errorf("failed to execute certificate executable: %w", err))
+			continue
+		}
+
+		for _, line := range resp.Lines.Item {
+			if strings.Contains(line, "NotAfter") {
+				// extract last part of line, which is the date contained within the parentheses
+				lineParts := strings.Split(line, " ")
+				if len(lineParts) < 1 {
+					continue
+				}
+
+				dateSection := lineParts[len(lineParts)-1]
+				date := strings.Trim(dateSection, "()")
+				t, err := time.Parse("060102150405Z", date)
+				if err != nil {
+					errs.AddPartial(1, fmt.Errorf("failed to parse validity date: %w", err))
+					continue
+				}
+
+				timeDifference := t.Unix() - now.AsTime().Unix()
```

Comment:
Use the built in [Time.Sub](https://pkg.go.dev/time#Time.Sub) function instead here.

Reply:
Updated. I then ran the scraper test excluding the pmetrictest.IgnoreMetricValues() and got similar values,  expected: 468123480, actual: 468045431 👍